frontend/tools/translationKey/generateTranslationKeyType.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Recursive function which returns sorted array of flattened keys.
+ *
+ * For example `{a: 'abc', b: {c: 'def'}}` is returned as `['<prefix>.a', '<prefix>.b.c']`
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Maybe we could not use `any` but instead this:
```
interface Translations {
  [key: string]: Translations | string
} 
```
Cannot be a type, though, because of circular referencing.